x-pack/filebeat/input/netflow/netflow_test.go

```diff
@@ -60,6 +67,131 @@ type TestResult struct {
 	Flows []beat.Event `json:"events,omitempty"`
 }
 
+func newV2Context() (v2.Context, func()) {
+	ctx, cancel := context.WithCancel(context.Background())
+	return v2.Context{
+		Logger:      logp.NewLogger("netflow_test"),
+		ID:          "test_id",
+		Cancelation: ctx,
+	}, cancel
+}
+
+func TestNetFlow(t *testing.T) {
+	pcaps, err := filepath.Glob(filepath.Join(pcapDir, "*.pcap"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, file := range pcaps {
+		testName := strings.TrimSuffix(filepath.Base(file), ".pcap")
+
+		t.Run(testName, func(t *testing.T) {
+
+			configMap := conf.MustNewConfigFrom(mapstr.M{})
+			pluginCfg, err := conf.NewConfigFrom(configMap)
+			require.NoError(t, err)
+
+			netflowPlugin, err := Plugin(logp.NewLogger("netflow_test")).Manager.Create(pluginCfg)
+			require.NoError(t, err)
+
+			mockPipeline := &pipelinemock.MockPipelineConnector{}
+
+			ctx, cancelFn := newV2Context()
+			errChan := make(chan error)
+			go func() {
+				defer close(errChan)
+				errChan <- netflowPlugin.Run(ctx, mockPipeline)
+			}()
+
+			defer cancelFn()
+
+			clientSeenTries := 0
+			for clientSeenTries < 5 {
+				if len(mockPipeline.GetConnectedClients()) > 0 {
+					break
+				}
+				time.Sleep(1 * time.Second)
+				clientSeenTries++
+				continue
+			}
+
+			if clientSeenTries == 5 {
+				t.Fatal("client did not connect to pipeline")
+				return
+			}
```

Comment:
Because it's a test, 1s seems to long to wait. If I understood it correctly, this for loop here is just waiting for the goroutine started at L101 torun and start the netflow plugin, right?

If that's the case, why not use a smaller timeout, like 1ms?

Another option to the loop is to use [`require.Eventually`](https://pkg.go.dev/github.com/stretchr/testify/require#Eventually) to check on a small interval like 1ms.
```suggestion
			require.Eventually(t, func() bool {
				return len(mockPipeline.GetConnectedClients()) > 0
			}, 5*time.Second, time.Millisecond, "client did not connect to pipeline")
```

Reply:
@belimawr I could argue the same question why wait for a shorter time!?